README.md

```diff
@@ -168,10 +168,9 @@ compilers:
   - &cpp 'g++ --std=c++11 $@ -o $@.exec'
 
 files:
-  - [ types.cpp, *cpp ]
-
-tests:
-  - [ types.cpp, $@.exec ]
+  - file: types.cpp
+    commands: *cpp
+    tests: $@.exec
```

Comment:
The only thing I worry about is that I seem to remember some caees where it would be non-obvious what file a test belonged under.

As long as we support the old syntax, I'm fine with that.

Also, what if you need multiple tests for a given file?

Reply:
The `tests` tag supports having a string or an array of strings just like the `commands` tag.

I was having trouble getting the two syntaxes to combine into one iterable array but I'll put a little more work into it later today

Reply:
Do you think we could get the wiki feature of GitHub turned on so we could go more in depth on specific parts, such as all the options for specs, without having an excessively long README?

Reply:
Idk. I can, but I like having the docs versioned in the repo. 

What about making a `docs/` folder in the repo, with markdown files inside?

Reply:
That sounds like a good idea. I'll start working on that in a separate PR